src/uct/ib/rc/accel/rc_mlx5_common.c

```diff
@@ -206,6 +260,76 @@ unsigned uct_rc_mlx5_iface_srq_post_recv_ll(uct_rc_mlx5_iface_common_t *iface)
     return count;
 }
 
+unsigned uct_rc_mlx5_iface_srq_post_recv_sge(uct_rc_mlx5_iface_common_t *iface)
+{
+    uct_ib_mlx5_srq_t *srq   = &iface->rx.srq;
+    uct_rc_iface_t *rc_iface = &iface->super;
+    uct_ib_mlx5_srq_seg_t *seg;
+    uint16_t count = 0, wqe_index, next_index;
+
+    uct_rc_mlx5_iface_srq_post_recv_check_union(iface);
+
+    wqe_index = srq->ready_idx;
+    for (;;) {
+        next_index = wqe_index + 1;
+        seg        = uct_ib_mlx5_srq_get_wqe(srq, next_index);
+        if (UCS_CIRCULAR_COMPARE16(next_index, >, srq->free_idx)) {
+            if (!seg->srq.free) {
+                break;
+            }
+
+            ucs_assert(next_index == (uint16_t)(srq->free_idx + 1));
```

Comment:
1. assertv
2. unite with other function by flag param

Reply:
The assert is not my doing... (Alex Mikheev 6 years ago)